contracts/adapters/linkswap/LinkswapStakingAdapter.sol

```diff
@@ -0,0 +1,176 @@
+
+    /**
+     * @return Amount of staked tokens / rewards earned after staking for a given account.
+     * @dev Implementation of ProtocolAdapter interface function.
+     */
+    function getBalance(address token, address account) external view override returns (uint256) {
+        if (token == YFL) {
+            uint256 totalRewards = 0;
+
+            totalRewards += ERC20(YFL).balanceOf(account);
+            totalRewards += ERC20(GOVERNANCE).balanceOf(account);
+            
+            totalRewards += StakingRewards(LSLP_YFL_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_YFL_WETH_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_DPI_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_MASQ_WETH_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_BUSD_LINK_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_GSWAP_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_CEL_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_YAX_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_CFI_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_USDC_POOL).earned(account,0);
+            totalRewards += StakingRewards(LSLP_LINK_USDT_POOL).earned(account,0);
+
+            return totalRewards;
+        } else if (token == CFI) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(CFI).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_LINK_CFI_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == MASQ) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(MASQ).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_MASQ_WETH_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == DPI) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(DPI).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_DPI_LINK_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == CEL) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(CEL).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_LINK_CEL_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == YAX) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(YAX).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_LINK_YAX_POOL).earned(account,1);
+            return totalRewards;
+        } else if (token == GSWAP) {
+            uint256 totalRewards = 0;
+            totalRewards += ERC20(GSWAP).balanceOf(account);
+            totalRewards += StakingRewards(LSLP_LINK_GSWAP_POOL).earned(account,1);
+            return totalRewards;
```

Comment:
here, user's balances are added to the return value

it shouldn't be here as only locked tokens should be returned

Reply:
[This contract](https://etherscan.io/address/0x0389d755C1833C9b350d4E8B619Eae16deFc1CbA) is currently accruing LSLP tokens as governance fee rewards whilst the `YFLPurchaser` contract is being completed. Once functional, users will be able to claim these rewards which auto-buy YFL, however the LSLP are currently locked and will continue to be (even when functional) until someone initiates a claim. How would you recommend tracking this, if at all?